docs/writing_review_screen.md

```diff
@@ -0,0 +1,240 @@
+---
+id: writing_review_screen
+title: |
+  Writing a Review Screen
+sidebar_label: |
+  Review Screen
+slug: /review_screen
+---
+
+## Overview
+
+Writing a review screen is very important to docassemble interviews. As the "back" button undoes an interview's progress,
+it's critical that people can edit answers they made early in the interview, without having to re-enter things later.
+
+Review screens are also critical for interviews to be fully WCAG compliant; for legal applications like the AssemblyLine was born from,
+[WCAG 2.1 success criterion 3.3.4](https://www.w3.org/TR/WCAG21/#error-prevention-legal-financial-data) requires that
+
+> a mechanism is available for reviewing, confirming, and correcting information before finalizing the submission
+
+in short, requiring a review screen to be present at the end of your interview, that allows the user to correct the information they entered.
+
+### Step 0: Complete the rest of the interview!
+
+Large changes to an interview's logic will require large changes to the review screen as well. Therefore, it's best to have mostly completed
+your interview before starting to make the review screen.
+
+### Step 1: ALDashboard Review Screen Generator
```

Comment:
> I'd suggest explaining what the user's task is when making a review screen

Do you have anything in mind specifically here? I tried to cover the UX perspective above (users need a way to edit answers, and DA's "undo as back" doesn't cut it). I don't want this doc to scope creep to becoming a concepts explainer, technical guide, and a tutorial, I want it to be a tutorial first and foremost. And most of what the review generator doesn't do is in the technical details that are hard to understand until you've used / seen a review screen

Reply:
Understood. Tutorials are definitely helpful, and people need a place to get started. The review screen section of the main documentation is so hard to use though, that a general explainer and resource could be helpful. But understood if you don't want to make this that thing.

I was not thinking of a full API reference. Maybe something like this:

Review screens let your user change their mind about their previous answers. Like the list of fields in a `question` screen, one review screen can let the user review many answers. This list of answers goes under the heading `review` instead of `fields`. Each answer you want to let your user change includes:

* A button, with a label of your choice (we usually use "Edit")
* A field name or a list of field names that you want the user to revisit when they click the edit button
* Some preview of the user's current answer. It's common to repeat or paraphrase the question from the interview and show the answer next to the question.

You can do more advanced things with review screens, too:
* If you would like the user to be able to jump to another screen (perhaps to edit a list of items), you can make the field name point to another question screen, labeled with an  `event`. This strategy is usually used to edit lists of people with a `revisit` block.
* You can trigger follow-up code to run after the user makes a change. For example: you can trigger a code block to get recomputed, delete the value of another answer, or trigger another follow-up screen.